src/coprocessor/dag/batch/executors/util/heap.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2017 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cell::RefCell;
+use std::cmp::{self, Ordering};
+use std::collections::BinaryHeap;
+use std::sync::Arc;
+use std::usize;
+
+use crate::coprocessor::codec::batch::LazyBatchColumnVec;
+use crate::coprocessor::dag::expr::Result;
+
+const HEAP_MAX_CAPACITY: usize = 1024;
+
+pub struct SortRow {
+    table: Arc<RefCell<LazyBatchColumnVec>>,
```

Comment:
Is it possible to store only `&LazyBatchColumnVec`? Or, if not, is it possible to use `Rc`?